### PR TITLE
adaptivecpp: 24.02.0, new

### PR DESCRIPTION
**Pull request by ouankou**
Topic Description
-----------------

adaptivecpp: 24.02.0, new

Package(s) Affected
-------------------

adaptivecpp: 24.02.0

Security Update?
----------------

No

Build Order
-----------

```
#buildit adaptivecpp
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`
- [x] AArch64 `arm64`
- [x] LoongArch 64-bit `loongarch64`

**Secondary Architectures**

Architectural progress for secondary ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

**Experimental Architectures**

Architectural progress for experimental ports does not impede on merging of this topic.

- [x] MIPS R6 64-bit (Little Endian) `mips64r6el`

### CODE REVIEW

**Comment by jiegec** [comment]
Can build and run the example from https://github.com/AdaptiveCpp/AdaptiveCpp/blob/develop/doc/examples.md properly.

File: app-devel/adaptivecpp/autobuild/defines [hunk 1]
```diff
@@ -0,0 +1,17 @@
+PKGNAME=adaptivecpp
+PKGSEC=devel
+PKGDEP="boost gcc-runtime glibc llvm python-3"
+PKGDES="Independent, community-driven compiler for C++-based heterogeneous programming models"
```
Comment (MingcongBai) [inline app-devel/adaptivecpp/autobuild/defines]: Compiler for C++-based heterogeneous programming models

File: app-devel/adaptivecpp/autobuild/defines [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+ABTYPE=cmake
```
Comment (MingcongBai) [inline app-devel/adaptivecpp/autobuild/defines]: Try cmakeninja?

Comment (MingcongBai) [reply app-devel/adaptivecpp/autobuild/defines]: I would also keep it near CMAKE_AFTER.

File: app-devel/adaptivecpp/autobuild/defines [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+BUILDDEP="boost llvm python-3"
+
+CMAKE_AFTER=" \
+    -DACPP_CONFIG_FILE_GLOBAL_INSTALLATION=ON \
+    -DDEFAULT_TARGETS='omp.accelerated' \
+    -DWITH_ACCELERATED_CPU=ON \
+    -DWITH_CUDA_BACKEND=OFF \
+    -DWITH_LEVEL_ZERO_BACKEND=OFF \
+    -DWITH_OPENCL_BACKEND=OFF \
```
Comment (MingcongBai) [inline app-devel/adaptivecpp/autobuild/defines:14]: Why disable all these features?

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: app-devel/adaptivecpp/autobuild/defines (hunk 3 above)
Comment (ouankou) [reply app-devel/adaptivecpp/autobuild/defines:14]: Similar to LLVM's OpenMP GPU offloading capabilities, all these features require the availability of corresponding components (such as CUDA, ROCM, and OpenCL implementations) both at compile time and runtime. However, we cannot guarantee that these components will always be available or that they will match between the compile and runtime environments. The only component consistently available is LLVM OpenMP on the host. Users have to build other features from source on their own machines.

Additionally, SSCP operates on a JIT compilation flow, allowing a single binary to potentially function across multiple backends like OpenCL and CUDA without recompilation. However, we can only support one host backend due to the constraints, and no backend switching would occur. Therefore, it is better to utilize AOT compilation over JIT, which has more overhead.

File: app-devel/adaptivecpp/autobuild/defines (hunk 1 above)
Comment (ouankou) [reply app-devel/adaptivecpp/autobuild/defines]: Fixed.

File: app-devel/adaptivecpp/autobuild/defines (hunk 2 above)
Comment (ouankou) [reply app-devel/adaptivecpp/autobuild/defines]: Fixed. `cmakeninja` works fine.

File: app-devel/adaptivecpp/autobuild/defines (hunk 3 above)
Comment (MingcongBai) [reply app-devel/adaptivecpp/autobuild/defines:14]: Fair enough.

**Review by MingcongBai — APPROVED** [review]

**Pull request merged**